### ISSUE TITLE
KeePass cannot save config

### ISSUE THREAD

**Issue by dbeckwith** [NixOS/nixpkgs#138737]
### Describe the bug
The KeePass application is unable to save its configuration state because of FS permissions.

### Steps To Reproduce
1. Add `<nixos-unstable>.keepass` to `environments.systemPackages` in `configuration.nix`
2. Run KeePass
3. Close KeePass
4. Get an error as the application tries to save its state (see screenshot)

### Expected behavior
The application to close and save its state successfully.

### Screenshots
![image](https://user-images.githubusercontent.com/1279939/134055326-febcf537-a74d-453a-9a1d-0bdd14928f3c.png)

### Additional context
System is NixOS 21.05, but the issue is only present with the `unstable` version of the package. The stable version of `keepass` from the `21.05` channel does not have this issue.

### Notify maintainers

@AndrewMorsillo 
@obadz 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.64, NixOS, 21.05.3248.6120ac5cd20 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(sonaxaton): `""`
 - channels(root): `"nixos-21.05.3248.6120ac5cd20, nixos-unstable-21.11pre316684.79c444b5bde"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - keepass
# a list of nixos modules affected by the problem
module:
```

**Comment by dbeckwith** [comment]
Also tagging @r-ryantm who [according to git](https://github.com/NixOS/nixpkgs/commit/51ce9692ef438b18a90dd4350a456d50d63c6f65#diff-ec507d0a4ec856dcaf90bd11872c85b9f8b3307340674311ecb7b14e3fab15ac) made the version bump that likely introduced the issue.

**Comment by datMaffin** [comment]
Looks like there is a recent issue upstream that seems related: https://sourceforge.net/p/keepass/bugs/2119/

**Comment by dbeckwith** [comment]
From that issue:

> On Linux systems, the local configuration file is typically stored in '$XDG_CONFIG_HOME/KeePass' 

Why is the Nix package trying to save it in `/lib` in its store path?

**Comment by dbeckwith** [comment]
@ryantm your bot made the upgrade to this package that introduced the issue. Do you have any ideas here?

**Comment by dbeckwith** [comment]
Hmm okay. The original maintainers haven't responded either, so maybe I'll just dig into this myself.

**Comment by dbeckwith** [comment]
Looks like this has already been fixed in unstable by #148136
